signac/synced_collections/numpy_utils.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Define utilities for handling NumPy arrays."""
+
+import warnings
+
+try:
+    import numpy
+
+    NUMPY = True
+except ImportError:
+    NUMPY = False
+
+
+class NumpyConversionWarning(UserWarning):
+    """Warning raised when NumPy data is converted."""
+
+
+NUMPY_CONVERSION_WARNING = (
+    "Any numpy types provided will be transparently converted to the "
+    "closest base Python equivalents."
+)
+
+
+def _convert_numpy(data):
+    """Convert a numpy data types to the corresponding base data types.
+
+    0d numpy arrays and numpy scalars are converted to their corresponding
+    primitive types, while other numpy arrays are converted to lists. If data
+    not a numpy data type, this function is a no-op.
+    """
+    if NUMPY:
+        if isinstance(data, numpy.ndarray):
+            # tolist will return a scalar for 0d arrays, so there's no need to
+            # special-case that check. 1-element 1d arrays should remain
+            # arrays, i.e. np.array([1])->[1], not 1.
+            warnings.warn(NUMPY_CONVERSION_WARNING, NumpyConversionWarning)
+            return data.tolist()
+        elif isinstance(data, (numpy.number, numpy.bool_)):
+            warnings.warn(NUMPY_CONVERSION_WARNING, NumpyConversionWarning)
+            return data.item()
+    return data
+
+
+def _is_atleast_1d_numpy_array(data):
+    """Check if an object is a nonscalar numpy array.
+
+    The need to ignore 0d numpy arrays is in typical in the synced
+    collections framework, since >0d arrays are mapped to (synced) lists
+    while 0d arrays are mapped to scalars.
```

Comment:
This line doesn't make sense. Do you mean:
```suggestion
    The synced collections framework must detect 0d numpy arrays, since >0d
    arrays are mapped to (synced) lists while 0d arrays are mapped to scalars.
```